bundles/binding/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchBinding.java

```diff
@@ -202,4 +229,67 @@ public void updated(Dictionary<String, ?> config)
 		}
 	}
 
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		// find the EcoTouch binding for the itemName
+		EcoTouchTags tag = null;
+		for (EcoTouchBindingProvider provider : providers) {
+			try {
+				tag = provider.getTypeForItemName(itemName);
+				break;
+			} catch (Exception e) {
+			}
+		}
+
+		// consider special cases
+		if (tag == EcoTouchTags.TYPE_ADAPT_HEATING) {
+			double adapt = Double.parseDouble(command.toString());
+			adapt = (adapt + 2) * 2;
+			adapt = Math.max(0, adapt);
+			adapt = Math.min(8, adapt);
+			command = new DecimalType((int) adapt);
+		}
+
+		EcoTouchConnector connector = new EcoTouchConnector(ip, username,
+				password, cookies);
+		int value = 0;
+		switch (tag.getType()) {
+		case Analog:
+			value = (int) (Double.parseDouble(command.toString()) * 10);
+			break;
+		case Word:
+			if (command == OnOffType.ON)
+				value = 1;
+			else if (command == OnOffType.OFF)
+				value = 0;
+			else
+				value = Integer.parseInt(command.toString());
+			break;
+		case Bitfield:
+			try {
+				value = connector.getValue(tag.getTagName());
+				int bitmask = 1 << tag.getBitNum();
+				if (command == OnOffType.OFF
+						|| Integer.parseInt(command.toString()) == 0) {
+					value = value & ~bitmask;
+				} else
+					value = value | bitmask;
+			} catch (Exception e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
```

Comment:
please remove the comment and use proper logging of this Exception. At least on debug level you should provide some useful information